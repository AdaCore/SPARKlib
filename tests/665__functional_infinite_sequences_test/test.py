from test_support import no_crash, sparklib_exec_test
import os

contains_manual_proof = False
os.environ["SPARKLIB_BODY_MODE"] = "On"


if __name__ == "__main__":
    no_crash(sparklib=True, opt=["--no-inlining", "-P", "test.gpr"])
    sparklib_exec_test()
