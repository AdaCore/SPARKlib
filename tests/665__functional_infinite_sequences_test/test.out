inst.ads:3:13: info: implicit aspect Always_Terminates on "Equal" has been proved, subprogram will terminate
inst.ads:3:57: info: division check proved (CVC5: 1 VC)
inst.ads:3:69: info: division check proved (CVC5: 1 VC)
inst.ads:4:13: info: implicit aspect Always_Terminates on "Equivalent" has been proved, subprogram will terminate
inst.ads:4:62: info: division check proved (CVC5: 1 VC)
inst.ads:4:73: info: division check proved (CVC5: 1 VC)
test.adb:17:07: info: initialization of "X" proved
test.adb:19:07: info: precondition proved (CVC5: 1 VC)
test.adb:19:15: info: predicate check proved (Trivial: 1 VC)
test.adb:19:26: info: predicate check proved (Trivial: 1 VC)
test.adb:20:24: info: unrolling loop
test.adb:23:07: info: precondition proved (CVC5: 1 VC)
test.adb:23:15: info: predicate check proved (Trivial: 1 VC)
test.adb:23:26: info: predicate check proved (Trivial: 1 VC)
test.adb:27:07: info: initialization of "X" proved
test.adb:29:24: info: unrolling loop
test.adb:32:07: info: precondition proved (Z3: 1 VC)
test.adb:32:15: info: precondition proved (CVC5: 1 VC)
test.adb:36:07: info: initialization of "X" proved
test.adb:38:07: info: precondition proved (CVC5: 1 VC)
test.adb:38:15: info: predicate check proved (Trivial: 1 VC)
test.adb:38:28: info: predicate check proved (Trivial: 1 VC)
test.adb:39:24: info: unrolling loop
test.adb:42:07: info: precondition proved (CVC5: 1 VC)
test.adb:42:15: info: predicate check proved (Trivial: 1 VC)
test.adb:42:28: info: predicate check proved (Trivial: 1 VC)
test.adb:47:07: info: precondition proved (CVC5: 1 VC)
test.adb:47:15: info: predicate check proved (Trivial: 1 VC)
test.adb:47:23: info: predicate check proved (Trivial: 1 VC)
test.adb:51:07: info: initialization of "X" proved
test.adb:51:10: info: initialization of "Y" proved
test.adb:53:07: info: precondition proved (CVC5: 1 VC)
test.adb:54:24: info: unrolling loop
test.adb:57:07: info: precondition proved (CVC5: 1 VC)
test.adb:59:07: info: precondition proved (CVC5: 1 VC)
test.adb:62:07: medium: precondition might fail
test.adb:65:07: info: precondition proved (Z3: 1 VC)
test.adb:69:07: info: initialization of "X" proved
test.adb:69:10: info: initialization of "Y" proved
test.adb:71:07: info: precondition proved (CVC5: 1 VC)
test.adb:72:24: info: unrolling loop
test.adb:75:07: info: precondition proved (CVC5: 1 VC)
test.adb:78:07: info: precondition proved (Z3: 1 VC)
test.adb:81:07: medium: precondition might fail
test.adb:83:07: info: precondition proved (Z3: 1 VC)
test.adb:87:07: info: initialization of "X" proved
test.adb:87:10: info: initialization of "Y" proved
test.adb:89:07: info: precondition proved (CVC5: 1 VC)
test.adb:90:24: info: unrolling loop
test.adb:93:07: info: precondition proved (CVC5: 1 VC)
test.adb:95:07: info: precondition proved (Z3: 1 VC)
test.adb:98:07: medium: precondition might fail
test.adb:100:07: medium: precondition might fail
test.adb:102:07: medium: precondition might fail
test.adb:106:07: info: initialization of "X" proved
test.adb:106:10: info: initialization of "Y" proved
test.adb:108:07: info: precondition proved (CVC5: 1 VC)
test.adb:109:24: info: unrolling loop
test.adb:112:07: info: precondition proved (CVC5: 1 VC)
test.adb:114:07: info: precondition proved (altergo: 1 VC)
test.adb:117:07: medium: precondition might fail
test.adb:120:07: info: precondition proved (Z3: 1 VC)
test.adb:124:07: info: initialization of "X" proved
test.adb:126:07: info: precondition proved (CVC5: 1 VC)
test.adb:126:19: info: precondition proved (CVC5: 1 VC)
test.adb:126:32: info: predicate check proved (CVC5: 1 VC; Trivial: 1 VC)
test.adb:126:35: info: predicate check proved (CVC5: 1 VC; Trivial: 1 VC)
test.adb:127:23: info: unrolling loop
test.adb:130:07: info: precondition proved (CVC5: 1 VC)
test.adb:130:15: info: precondition proved (CVC5: 1 VC)
test.adb:130:28: info: predicate check proved (CVC5: 1 VC; Trivial: 1 VC)
test.adb:130:31: info: predicate check proved (CVC5: 1 VC; Trivial: 1 VC)
test.adb:131:07: info: precondition proved (Z3: 1 VC)
test.adb:131:15: info: precondition proved (CVC5: 1 VC)
test.adb:131:28: info: predicate check proved (CVC5: 1 VC; Trivial: 1 VC)
test.adb:131:31: info: predicate check proved (CVC5: 1 VC; Trivial: 1 VC)
test.adb:132:07: info: precondition proved (Z3: 1 VC)
test.adb:132:19: info: precondition proved (CVC5: 1 VC)
test.adb:132:32: info: predicate check proved (CVC5: 1 VC; Trivial: 1 VC)
test.adb:132:35: info: predicate check proved (CVC5: 1 VC; Trivial: 1 VC)
test.adb:133:07: info: precondition proved (Z3: 1 VC)
test.adb:133:19: info: precondition proved (CVC5: 1 VC)
test.adb:133:32: info: predicate check proved (CVC5: 1 VC; Trivial: 1 VC)
test.adb:133:35: info: predicate check proved (CVC5: 1 VC; Trivial: 1 VC)
test.adb:137:07: info: initialization of "X" proved
test.adb:139:07: info: precondition proved (CVC5: 1 VC)
test.adb:139:15: info: predicate check proved (Trivial: 1 VC)
test.adb:139:29: info: predicate check proved (Trivial: 1 VC)
test.adb:140:23: info: unrolling loop
test.adb:143:07: info: precondition proved (Z3: 1 VC)
test.adb:143:15: info: predicate check proved (Trivial: 1 VC)
test.adb:143:29: info: predicate check proved (Trivial: 1 VC)
test.adb:144:07: info: precondition proved (Z3: 1 VC)
test.adb:144:15: info: predicate check proved (Trivial: 1 VC)
test.adb:144:30: info: predicate check proved (Trivial: 1 VC)
test.adb:145:07: info: precondition proved (Z3: 1 VC)
test.adb:145:15: info: predicate check proved (Trivial: 1 VC)
test.adb:145:29: info: predicate check proved (Trivial: 1 VC)
test.adb:151:07: info: precondition proved (CVC5: 1 VC)
test.adb:151:15: info: predicate check proved (Trivial: 1 VC)
test.adb:151:26: info: predicate check proved (Trivial: 1 VC)
test.adb:155:07: info: initialization of "X" proved
test.adb:157:24: info: unrolling loop
test.adb:160:12: info: precondition proved (CVC5: 1 VC)
test.adb:160:20: info: predicate check proved (CVC5: 1 VC; Trivial: 1 VC)
test.adb:161:07: info: precondition proved (CVC5: 1 VC)
test.adb:161:15: info: precondition proved (CVC5: 1 VC)
test.adb:162:07: info: precondition proved (Z3: 1 VC)
test.adb:162:15: info: precondition proved (CVC5: 1 VC)
test.adb:162:34: info: precondition proved (CVC5: 1 VC)
test.adb:163:07: info: precondition proved (CVC5: 1 VC)
test.adb:163:15: info: predicate check proved (Trivial: 1 VC)
test.adb:163:26: info: predicate check proved (Trivial: 1 VC)
test.adb:167:07: info: initialization of "X" proved
test.adb:167:10: info: initialization of "Y" proved
test.adb:174:33: info: loop invariant initialization proved (CVC5: 1 VC)
test.adb:174:33: info: loop invariant preservation proved (CVC5: 1 VC)
test.adb:174:33: info: predicate check proved (Trivial: 2 VC)
test.adb:175:33: info: loop invariant initialization proved (CVC5: 1 VC)
test.adb:175:33: info: loop invariant preservation proved (CVC5: 1 VC)
test.adb:175:56: info: precondition proved (CVC5: 2 VC)
test.adb:177:07: info: precondition proved (CVC5: 1 VC)
test.adb:177:15: info: predicate check proved (Trivial: 1 VC)
test.adb:177:26: info: predicate check proved (Trivial: 1 VC)
test.adb:178:07: info: precondition proved (CVC5: 1 VC)
test.adb:178:15: info: precondition proved (CVC5: 1 VC)
test.adb:179:07: info: precondition proved (CVC5: 1 VC)
test.adb:179:15: info: precondition proved (CVC5: 1 VC)
test.adb:182:12: info: precondition proved (CVC5: 1 VC)
test.adb:182:20: info: predicate check proved (CVC5: 1 VC; Trivial: 1 VC)
test.adb:183:07: info: precondition proved (CVC5: 1 VC)
test.adb:183:15: info: precondition proved (CVC5: 1 VC)
test.adb:184:07: info: precondition proved (CVC5: 1 VC)
test.adb:184:15: info: precondition proved (CVC5: 1 VC)
test.adb:185:07: info: precondition proved (CVC5: 1 VC)
test.adb:185:15: info: precondition proved (CVC5: 1 VC)
test.adb:186:07: info: precondition proved (CVC5: 1 VC)
test.adb:186:15: info: predicate check proved (Trivial: 1 VC)
test.adb:186:26: info: predicate check proved (Trivial: 1 VC)
test.adb:190:07: info: initialization of "X" proved
test.adb:192:24: info: unrolling loop
test.adb:195:12: info: precondition proved (CVC5: 1 VC)
test.adb:195:23: info: predicate check proved (CVC5: 1 VC; Trivial: 1 VC)
test.adb:196:07: info: precondition proved (CVC5: 1 VC)
test.adb:196:15: info: predicate check proved (Trivial: 1 VC)
test.adb:196:26: info: predicate check proved (Trivial: 1 VC)
test.adb:197:07: info: precondition proved (Z3: 1 VC)
test.adb:197:15: info: precondition proved (CVC5: 1 VC)
test.adb:198:07: info: precondition proved (Z3: 1 VC)
test.adb:198:15: info: precondition proved (Z3: 1 VC)
test.adb:202:07: info: initialization of "X" proved
test.adb:204:24: info: unrolling loop
test.adb:207:07: info: precondition proved (Z3: 1 VC)
test.adb:208:07: info: precondition proved (Z3: 1 VC)
test.adb:212:23: info: all elements shall fit in index type for sequence aggregates at spark-containers-functional-infinite_sequences.ads:61, in instantiation at inst.ads:5
test.adb:212:23: info: precondition proved (Trivial: 1 VC)
test.adb:214:07: info: precondition proved (CVC5: 1 VC)
test.adb:214:15: info: predicate check proved (Trivial: 1 VC)
test.adb:214:26: info: predicate check proved (Trivial: 1 VC)
test.adb:215:07: info: precondition proved (CVC5: 1 VC)
test.adb:215:15: info: precondition proved (CVC5: 1 VC)
test.adb:215:34: info: precondition proved (CVC5: 1 VC)
